### PR TITLE
Allows set downloadable permissions to any downloadable product in any order

### LINKED ISSUE #29884: Cant add additional download permissions on 5.3
**Prerequisites (mark completed items with an [x]):**
- [ ] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [ ] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

### PR DESCRIPTION
**Pull request by claudiosanches**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This restores capability to set downloadable permissions to any downloadable product, no matter if this product got bought or not... For me this isn't a good idea and sounds very unsafe, since it's impossible to keep track of those downloadable permissions properly since no validation is done, but users are requesting to remove this validation, they don't seems to care about keep track or maintain safe, so here we go.

Closes #29884

### How to test the changes in this Pull Request:

1. Create an order from either within the dashboard or normal checkout that does not include the product we are going to manually grant access to later
2. Make sure you've fully saved the order
3. Enter the downloadable product name into the 'Downloadable product permissions' input
4. Click 'Grant access' - you'll see the notice Could not grant access.
5. Now apply this patch and try to click in 'Grant access', it will works as before in Woo 5.1 or 5.2.2

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Fix - Restored behavior that allows downloadable product to have permissions set in any order.

### CODE REVIEW

**Comment by jakewhiteley** [comment]
@claudiosanches I see this from both perspectives and would like to propose a better solution that keeps everyone happy.

Here is the root issue:

* people want to be able to freely assign permissions to users, regardless of if they have purchased the item
* previously this required a placeholder order to be created
* currently an order has to be created with the item being recorded as purchased (if only temporarily)

And here are the side effects:
* people who use ERPs don't want a $0.00 order being recorded most of the time when assigning a permission
* people don't want to trigger emails for an order when granting permissions
* In both situations (pre and post 5.3) managing permissions is difficult

### Proposed solution
I propose that a new dashboard screen is created where admins can view/assign/manage download permissions **completely separate** from orders.

I propose the current 5.3 way of adding permissions via an order stays - it makes sense an order must contain the product you are granting permissions for, and the use cases are few and far between (it's probably just to re-grant permissions that have expired?).

But admins who want to grant permissions to customers that haven't purchased should do so via the new permissions overview screen - thus sidestepping a redundant order creation, and keeping it "safe" and tracked.

I am happy to create a new feature request for this admin screen.

**Comment by claudiosanches** [comment]
Hello @jakewhiteley 

> I propose that a new dashboard screen is created where admins can view/assign/manage download permissions completely separate from orders.

Sorry, but this sounds even worse for me, still you don't have capability track it per order, track discounts and free downloads.

**Review by jonathansadowski — APPROVED** [review]
Nice work — verified that this restores the behavior to what it was prior to the release of 5.3. 👍

**Comment by jakewhiteley** [comment]
@claudiosanches the new dashboard screen would cover all that. It would be a proper downloads management system.

An admin should be able to see when it was granted, how it was granted (via an order or manually), who granted it (if applicable), expiry date, and how many times the item has been used on a grant by grant basis.

Ideally some of this info would be presented on other screens, such as download count and a way to find granted users directly from the product screen, and the ability to see all grants from the user edit screen 

Alternatively, the downloads screen would have different views to present this info ( by grant, by user, by downloadable product).

Basically, neither the current nor the previous solution are particularly useful, and I personally think a proper overhaul could allow for a much better way of managing downloads.

**Comment by BonnieBeeman** [comment]
I also give strong support for re-enabling the former behavior. Very obviously this is a very common and welcome use case. The people who don't use it won't see a difference and the people who do us it are missing it dearly.  Adding a product to any order allows us to provide special products to specific people and the product shows up in "My Account".  Best option would be if the Admin could update a set of accounts at the same time with a new product, or replace the product that user was given permission to access.  Thank you for re-enabling this extremely important feature.  Sorry it didn't make it into version 5.5.0

**Comment by BonnieBeeman** [comment]
Tried code for Downloadable Permissions and it works as-expected.  
Many thanks.
